lib/Serialization/SerializedModuleLoader.cpp

```diff
@@ -547,6 +557,10 @@ FileUnit *SerializedModuleLoaderBase::loadAST(
     loadInfo.status =
         loadedModuleFile->associateWithFileContext(fileUnit, diagLocOrInvalid,
                                                    treatAsPartialModule);
+    if (auto shadowded = loadedModuleFile->getShadowedModule())
+      if (shadowded->isSystemModule())
```

Comment:
This seems wrong—whether or not a Swift overlay is a system module isn't *really* dependent on whether the underlying thing is a system module. Is this needed, given that we treat the resource search paths as system search paths now?

Also, typo: "shadowded".

Reply:
Is it possible for users to create their own 'user' overlays of system frameworks ?

Reply:
I'm not sure it's *really* needed. I added it just *not* to change the behavior. Otherwise bunch of tests fails.

Reply:
> Is it possible for users to create their own 'user' overlays of system frameworks ?
 
Possible, yes; supported, not really. And in that case, calling them "system" would be misleading.

I guess since this is the current behavior we can fix it later. Can you leave a comment, at least?

Reply:
> Can you leave a comment, at least?

Sure. And I'll fix typo.